urdf/CMakeLists.txt

```diff
@@ -45,6 +45,10 @@ link_directories(${Boost_LIBRARY_DIRS} ${catkin_LIBRARY_DIRS})
 add_library(${PROJECT_NAME} src/model.cpp src/rosconsole_bridge.cpp)
 target_link_libraries(${PROJECT_NAME} ${TinyXML_LIBRARIES} ${catkin_LIBRARIES} ${urdfdom_LIBRARIES})
 
+if(WIN32)
+  target_compile_definitions(${PROJECT_NAME} PRIVATE "URDF_BUILDING_DLL")
```

Comment:
Since this is `kinetic-devel` which targets [cmake 3.0.2](http://www.ros.org/reps/rep-0003.html) how about using [GenerateExportHeader](https://cmake.org/cmake/help/v3.0/module/GenerateExportHeader.html)?

Reply:
So, the documentation for GenerateExportHeader says it is deprecated :).  I guess I'd rather not switch from one deprecated thing to another.  Unless you think this is really important, I think I'll leave it for now.

On a different but related note, the define was actually wrong.  The code expected `URDF_BUILDING_LIBRARY`, but cmake was setting `URDF_BUILDING_DLL`.  I've fixed that now in 888b4fd (but nothing will show up in CI because it is only for Windows).

Reply:
I think the "This function is deprecated" at the bottom is referring to `add_compiler_export_flags`

Reply:
Yeah, you are right about that, sorry.  I will point out here that while we could make this change, it is actually somewhat difficult to test in ROS1 (since we have no Windows build to test with).  While we could test this in ROS2 and then backport it here to keep the diff down, I think I'd rather do that separately (if it seems like I am trying to avoid that work right now, I am :).  Is it OK if we do it that way instead?  I can open an issue about it.

Reply:
Sure, opening an issue for it sounds good to me. It might be a good one to put `help wanted` on it since as long as what's here works we're unlikely to spend time on it.

Reply:
Done in #11